src/main/java/org/jenkinsci/plugins/codesonar/conditions/NewWarningsIncreasedByPercentageCondition.java

```diff
@@ -60,18 +61,26 @@ public Result validate(CodeSonarBuildActionDTO current, CodeSonarBuildActionDTO
         Analysis currentActiveWarnings = current.getAnalysisActiveWarnings();
         Analysis currentNewWarnings = current.getAnalysisNewWarnings();
 
-        float activeWarningCount = (float) currentActiveWarnings.getWarnings().size();
-        float newWarningCount = (float) currentNewWarnings.getWarnings().size();
+        int activeWarningCount = currentActiveWarnings.getWarnings().size();
+        int newWarningCount = currentNewWarnings.getWarnings().size();
 
         float result = (newWarningCount * 100.0f) / activeWarningCount; 
+        float thresholdPercentage = Float.parseFloat(percentage);
 
-        if (result > Float.parseFloat(percentage)) {
+        if (result > thresholdPercentage) {
+            resultDescription = String.format("More than %.2f%% new warnings (%.2f%%, %d out of %d)", thresholdPercentage, result, newWarningCount, activeWarningCount);
             return Result.fromString(warrantedResult);
         }
 
+        resultDescription = String.format("At most %.2f%% new warnings (%.2f%%, %d out of %d)", thresholdPercentage, result, newWarningCount, activeWarningCount);
```

Comment:
@ampleyfly After looking at these proposed result descriptions in more detail, We thought that it would be clearer to show the name of the condition followed by the results, rather than replacing the name of the condition with the results.  We also felt that showing the condition parameter names would enhance the clarity.

Your proposed result descriptions would yield text on the build page similar to the following:

> More than 1.00% new warnings (1.2%, 12 out of 1000)

I propose we display the text like the following instead:

> Warning count increase: new only [threshold=1.00%, increase=1.20% (count: new=12, total=1000)]

Do you think this proposed text is good too?  Do you still prefer your original proposal?  Would you make further changes to my proposal here?

Honestly, I might be inclined to make further changes to this text myself, but only to make the notation cleaner.

Reply:
I agree that your suggestion looks better :)
I can make that change if you'd like.

Reply:
Great, thanks for your feedback.  We plan to make the updates on PR #10 since we think trying to merge both this PR and PR #10 will make plenty of merge conflicts.  We will also test the changes before merging.  There is no need for you to make the changes on your fork/branch.

Reply:
@ampleyfly below is an example screenshot from PR #10 (somewhat trivial results, I know), but it should show you how the results will look:

![codesonar_plugin build_results 20230405](https://user-images.githubusercontent.com/62718984/230080319-fc53d573-c308-47b3-bc4a-b7a54ebaeb9a.png)

Does this look good to you?

Reply:
Yes!

Reply:
Great, thanks!  We think we are close to merging and releasing our changes.  If the remaining tests go well, then the release should be available in the next few days.